lib/src/main/java/com/diffplug/spotless/npm/NpmProcess.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2016-2020 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.npm;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+class NpmProcess {
+
+	private final File workingDir;
+
+	private final File npmExecutable;
+
+	NpmProcess(File workingDir, File npmExecutable) {
+		this.workingDir = workingDir;
+		this.npmExecutable = npmExecutable;
+	}
+
+	void install() {
+		npmAwait("install", "--no-audit", "--no-package-lock");
```

Comment:
I don't want to make things too complicated, but I've been really happy with `npm ci`.  On newer versions of npm, it runs *way* faster than `npm install`, since it relies exclusively on the precomputed `package-lock.json`.  Doesn't have to be part of this, I'm not sure how to document it, but it might be possible to speed things up a fair bit using `npm ci` at some point in the future.

Reply:
Totally. And it might even be a good idea to include the used packages in our distribution instead of downloading them repeatedly. This might be achieved using something like [npm-bundle](https://github.com/majgis/npm-bundle)...

Reply:
So would you say we keep that in mind for future improvement?

If this PR gets merged, we might also consider to bump the default versions of the used node-based formatters to their latest version.

Reply:
OK, versions are bumped.